src/plugin/Plugin.h

```diff
@@ -1065,6 +1066,28 @@ class Plugin
 	 */
 	virtual bool HookQueueEvent(Event* event);
 
+	/**
+	 * Hook into dispatching events. Whenever a queued event is about to be
+	 * executed, it first gives all plugins with this hook enabled a chance to
+	 * handle the execution otherwise. A plugin can either just inspect the
+	 * event, or take it over (i.e., prevent the event manager from executing
+	 * the event handler).
+	 *
+	 * The default implementation never handles the dispatching in any way.
+	 *
+	 * @param event The event to be dispatched. The method can modify it in
+	 * place as long as it ensures matching types and correct reference
+	 * counting.
+	 *
+	 * @param no_remote True if the event should not be auto-published to
+	 * remote peers (e.g., if the event itself was received from remote).
+	 *
+	 * @return True if the plugin took charge of the event; in that case it
+	 * must have assumed ownership of the event and the event manager will not
+	 * do anything further with it. False otherwise.
```

Comment:
This touches the same ownership question I mentioned earlier — just flagging here too.